### PR TITLE
Cleanup memory::blob_manager.

### PR DESCRIPTION
**Pull request by szmyd**

### CODE REVIEW

File: src/lib/memory/blob_manager.cpp [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 namespace homeobject {
 
-ShardIndex& MemoryHomeObject::_find_index(shard_id shard) {
+ShardIndex& MemoryHomeObject::_find_index(shard_id shard) const {
```
Comment (xiaoxichen) [inline src/lib/memory/blob_manager.cpp:5]: Oh the tricks here is use unique_ptr(or any ptr) so the const is on ptr

File: src/lib/memory/blob_manager.cpp [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
     auto index_it = _in_memory_index.find(shard);
     RELEASE_ASSERT(_in_memory_index.end() != index_it, "Missing BTree!!");
     return *index_it->second;
```
Comment (xiaoxichen) [inline src/lib/memory/blob_manager.cpp:8]: NPE here if shard doesnt exists

File: src/lib/memory/blob_manager.cpp [hunk 3]
```diff
@@ -12,62 +12,69 @@ BlobManager::Result< blob_id > MemoryHomeObject::_put_blob(ShardInfo const& shar
     // Lookup Shard Index (BTree and SSN)
     auto& our_shard = _find_index(shard.id);
 
-    // Lock the BTree
-    // TODO: Generate BlobID (with RAFT this will happen implicitly) and Route
-    auto route = BlobRoute{shard.id, our_shard._shard_seq_num++};
+    // Generate BlobID (with RAFT this will happen implicitly) and Route
+    auto const route = BlobRoute{shard.id, our_shard._shard_seq_num++};
+    if ((UINT64_MAX >> 16) < route.blob) { return folly::makeUnexpected(BlobError::UNKNOWN); }
```
[the remark this answers is not in the document]
Comment (szmyd) [reply src/lib/memory/blob_manager.cpp:17]: Hmmm, yes. I think I was thinking shard good catch. Although no modern machine has >48 bits of addressable memory anyways. Will fix next time I'm in that code.

File: src/lib/memory/blob_manager.cpp (hunk 2 above)
Comment (szmyd) [reply src/lib/memory/blob_manager.cpp:8]: NPE?

File: src/lib/memory/blob_manager.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (szmyd) [reply src/lib/memory/blob_manager.cpp:17]: Still waiting in a change to make the Blob's body movable outside the unique_ptr; so can remove this check then.

File: src/lib/memory/blob_manager.cpp (hunk 2 above)
Comment (xiaoxichen) [reply src/lib/memory/blob_manager.cpp:8]: nevermind,  you catch that by RELEASE_ASSERT.

File: src/lib/memory/blob_manager.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (xiaoxichen) [reply src/lib/memory/blob_manager.cpp:17]: yes that part looks good

**Pull request merged**